controller/array_action/array_mediator_interface.py

```diff
@@ -243,13 +232,13 @@ def get_object_by_id(self, object_id, object_type):
         raise NotImplementedError
 
     @abstractmethod
-    def create_snapshot(self, name, volume_name, pool_id=None):
+    def create_snapshot(self, volume_id, snapshot_name, pool=None):
         """
         This function should create a snapshot from volume in the storage system.
         Args:
-            name           : name of the snapshot to be created in the storage system
-            volume_name    : name of the volume to be created from
-            pool_id: pool_id of the volume to find the snapshot more efficiently.
+            volume_id : id of the volume to be created from
+            snapshot_name : name of the snapshot to be created in the storage system
+            pool : pool to create the snapshot in
         Returns:
             Snapshot
         Raises:
```

Comment:
should xiv `create_snapshot` raise `SnapshotSourcePoolMismatch` as well?

Reply:
it's only for `create_snapshot ` (and yes, it was in `create_volume` by mistake, it might go there in the ds8k ticket but not in this scope)